### ISSUE TITLE
The swap code in array destructuring gives an error

### ISSUE THREAD

**Issue by kimsk** [basarat/typescript-book#233]
In the [array destructuring](https://github.com/basarat/typescript-book/blob/master/docs/destructuring.md#array-destructuring) section, there is a section of code:

```typescript
    var x = 1, y = 2;
    [x, y] = [y, x];
    console.log(x, y); // 2,1
```

Somehow I got the error:

    TSError: ⨯ Unable to compile TypeScript: Left side of comma operator is unused and has no side effects. (2695)

Am I missing something?

**Comment by kimsk** [comment]
I forgot the `;` at the end of the first line. Sorry for the false alarm.
